ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/transaction/filtered/UtxoFilteredTransaction.java

```diff
@@ -124,4 +126,12 @@ public interface UtxoFilteredTransaction extends TransactionWithMetadata {
      * @throws CordaRuntimeException if the current {@link UtxoFilteredTransaction} fails to verify correctly.
      */
     void verify();
+
+    /**
+     * Retrieve the Merkle proof for each component group.
+     *
+     * @return A map containing component group IDs as keys and Merkle proofs as values.
+     */
+    @NotNull
+    Map<Integer, MerkleProof> getMerkleProofs();
```

Comment:
How about having this take a set of ints as argument and only returning `MerkleProof` for the specificed component groups?

Reply:
Yeah that might be a better approach.

Reply:
Moving this to the internal API instead, I don't think this needs to be public after giving it some thoughts.